hazelcast/src/main/java/com/hazelcast/nio/serialization/DefaultPortableReader.java

```diff
@@ -299,19 +331,22 @@ private int readNestedPosition(String fieldName, FieldType type) throws IOExcept
     }
 
     private int readPosition(FieldDefinition fd) throws IOException {
-        return in.readInt(offset + fd.getIndex() * 4);
+        int pos = in.readInt(offset + fd.getIndex() * Bits.INT_SIZE_IN_BYTES);
+        short len = in.readShort(pos);
+        // name + len + type
+        return pos + Bits.SHORT_SIZE_IN_BYTES + len + 1;
```

Comment:
Ok I get that when I start to read setPosition method. len is the size of the field name. All clear.